bigquery_etl/cli/udf.py

```diff
@@ -266,3 +269,74 @@ def publish(ctx, path, project, dependency_dir, gcs_bucket, gcs_path):
 
 
 mozfun.add_command(publish)
+
+
+@udf.command(
+    help="Rename UDF or UDF dataset.",
+)
+@click.argument("name", required=True)
+@path_option
+@click.option("--new_name", "-n", help="New UDF or dataset name", required=True)
+@click.option(
+    "--sql_path",
+    "--sql-path",
+    help="Path to directory with SQL queries.",
+    type=click.Path(file_okay=False),
+    default="sql/",
+)
+@click.pass_context
+def rename(ctx, name, path, new_name, sql_path):
+    """Rename UDFs based on pattern."""
+    udf_dirs = ctx.obj["UDF_DIRS"]
+    if path and is_valid_dir(None, None, path):
+        udf_dirs = (path,)
+
+    if UDF_NAME_RE.match(name) and UDF_NAME_RE.match(new_name):
+        # rename UDF
+        udf_files = _udfs_matching_name_pattern(name, udf_dirs)
+        match = UDF_NAME_RE.match(new_name)
+        new_udf_name = match.group("name")
+        new_udf_dataset = match.group("dataset")
+    elif UDF_DATASET_RE.match(name) and UDF_DATASET_RE.match(new_name):
```

Comment:
Should we instead use globbing to indicate renaming a dataset? `bqetl rename from-dataset.* to-dataset`. This may be simpler to reason about since it's similar to `mv`